plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -512,9 +512,9 @@ public PartitionStatistics getTableStatistics(Table table)
         PartitionStatistics missingColumnStatistics = delegate.getTableStatistics(tableWithOnlyMissingColumns);
 
         // merge missing stats into the cached stats
-        return tableStatisticsCache.asMap().compute(
-                tableName,
-                (ignored, oldStats) -> {
```

Comment:
I think it would be cleaner to have `get` that accepts `merger` (similar to something like https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html#merge(K,V,java.util.function.BiFunction ). This way invalidation problem (essentially cache mechanism) would be separated from stats merging problem.

Reply:
That makes sense, I extracted loading and merging logic in both cases